api/src/main/java/org/apache/iceberg/BatchScanAdapter.java

```diff
@@ -140,4 +140,8 @@ public int splitLookback() {
   public long splitOpenFileCost() {
     return scan.splitOpenFileCost();
   }
+
+  public Scan getScanDelegate() {
```

Comment:
it's not clear why is this change is needed. Could you elaborate please?

Reply:
SnapShotScan has the relevant read metrics. Exposing the backing scan object from the adapter to extract the metrics. (which will be used [here](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/Scan.java#L126))

Reply:
I believe this no longer applies, let me resolve this thread.